[JavaScript] Add new concept excercise: array-destructuring (#2536)
exercism/v3: languages/javascript/exercises/concept/array-destructuring/.docs/instructions.md

Elyse, magician-to-be, continues her training. She has a deck of cards she wants to manipulate.

To make things easier, she usually only starts with cards numbered 1 to 10, although some of the tricks may involve additional cards.

## 1. Get the first card

> Elyse will summon the first card in the deck.

Reviewer: ```suggestion
Elyse will summon the first card in the deck, without using the `array[index]` or `.shift()`. It's just like magic.
```

```javascript
const deck = [5, 9, 7, 1, 8]

getFirstCard(deck)
// => 5
```

## 2. Get the second card

> Elyse will summon the second card in the deck.

Reviewer: ```suggestion
Elyse performs sleight of hand, and summons the second card in the deck, without using the `array[index]`.
```

```javascript
const deck = [3, 2, 10, 6, 7]

getSecondCard(deck)
// => 2
```

## 3. Swap the first two cards

> Elyse will make the top two cards of the deck switch places.

Reviewer: ```suggestion
Elyse will make the top two cards of the deck switch places. She doesn't need to call a single function.
```
